### PR TITLE
[SPARK-34927][INFRA] Support TPCDSQueryBenchmark in Benchmarks

### PR DESCRIPTION
**Pull request by MyeongKim**
### What changes were proposed in this pull request?

This PR adds a new option(run-tcp-ds-query-benchmark) in 'Run benchmarks' workflow of Github actions in order to support TPCDSQueryBenchmark. 
When this option is true, the job will populate a specific matrix with `org.apache.spark.sql.execution.benchmark.TPCDSQueryBenchmark` class and `"--data-location tpcds-sf-1"` argument which then executes the TPCDSQueryBenchmark class with the specific argument. Other benchmark classes runs in parallel from separate jobs.

To set up TPC-DS (SF=1) table data, a previous [merged PR](https://github.com/apache/spark/pull/31886/files) was referred 

(NOTE: Also, we need to update 'Running benchmarks in your forked repository' section of  https://spark.apache.org/developer-tools.html to address the new option.)

### Why are the changes needed?

To support a new benchmark class(TPCDSQueryBenchmark.scala) via Github Actions. Currently [this class is excluded](https://github.com/apache/spark/blob/71effba5f2f6f6e4755d0af3411d8d15d0e24577/core/src/test/scala/org/apache/spark/benchmark/Benchmarks.scala#L100) from this feature because the current structure doesn't take an argument for a specific benchmark class. 

### Does this PR introduce _any_ user-facing change?

No. It's only for developers.

### How was this patch tested?

1. Manually tested in my fork repository
- [without TPCDSQueryBenchmark(the new option value equals to 'false')](https://github.com/MyeongKim/spark/actions/runs/1065162581)
- [with TPCDSQueryBenchmark(the new option value equals to 'true')](https://github.com/MyeongKim/spark/actions/runs/1065365313)

2. More test cases with the combinations below

| input class string | run-tcp-ds-query-benchmark flag | GA url |
| ------------- | ------------- | ------------- |
| org.apache.spark.ml.linalg.BLASBenchmark | false |  https://github.com/MyeongKim/spark/actions/runs/1065165135 |
| org.apache.spark.ml.linalg.BLASBenchmark | true | https://github.com/MyeongKim/spark/actions/runs/1065369228 |
| org.apache.spark.sql.* | false | https://github.com/MyeongKim/spark/actions/runs/1066119336 |
| org.apache.spark.sql.* | true | https://github.com/MyeongKim/spark/actions/runs/1066398591 |

3. GA produces an expected `TPCDSQueryBenchmark-results.txt` result file.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
ok to test

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -19,32 +19,107 @@ on:
...
     steps:
     - name: Generate matrix
       id: set-matrix
-      run: echo "::set-output name=matrix::["`seq -s, 1 $SPARK_BENCHMARK_NUM_SPLITS`"]"
+      run: |
+        echo "::set-output name=class::['"$SPARK_BENCHMARK_CLASS"']"
+        echo "::set-output name=split::["`seq -s, 1 $SPARK_BENCHMARK_NUM_SPLITS`"]"
+        echo "::set-output name=num-split::['"$SPARK_BENCHMARK_NUM_SPLITS"']"
+
+  generate-TCP-DS-dataset:
```
Comment (HyukjinKwon) [inline .github/workflows/benchmark.yml:46]: Can you leave a cross comment in build_and_test.yml so we don't forgot to update together?

Comment (HyukjinKwon) [inline .github/workflows/benchmark.yml:46]:
```suggestion
  generate-tpc-ds-dataset:
```

**Comment by HyukjinKwon** [comment]
ok to test

File: .github/workflows/benchmark.yml (hunk 1 above)
Comment (MyeongKim) [reply .github/workflows/benchmark.yml]: thank you for your comment! Addressed in an additional commit

Comment (MyeongKim) [reply .github/workflows/benchmark.yml]: Thanks! Addressed in an additional commit

File: .github/workflows/benchmark.yml [hunk 2]
```diff
@@ -19,32 +19,108 @@ on:
...
 
 jobs:
   matrix-gen:
     name: Generate matrix for job splits
     runs-on: ubuntu-20.04
     outputs:
-      matrix: ${{ steps.set-matrix.outputs.matrix }}
+      class: ${{ steps.set-matrix.outputs.class }}
+      split: ${{ steps.set-matrix.outputs.split }}
+      num-split: ${{ steps.set-matrix.outputs.num-split }}
```
Comment (HyukjinKwon) [inline .github/workflows/benchmark.yml:34]: I think you won't have to pass it via GitHub outputs. `matrix` is being passed to parse a multi element JSON array but for `class` and `num-split`, we could just directly access it by `${{ steps.set-matrix.outputs.class }}` or  `${{ steps.set-matrix.outputs.num-split }}`.

File: .github/workflows/benchmark.yml [hunk 3]
```diff
@@ -19,32 +19,108 @@ on:
...
+    name: "Run benchmarks: ${{ matrix.class }} (JDK ${{ github.event.inputs.jdk }}, ${{ matrix.split }} out of ${{ github.event.inputs.num-splits }} splits)"
+    if: always()
+    needs: [matrix-gen, generate-tpc-ds-dataset]
     # Ubuntu 20.04 is the latest LTS. The next LTS is 22.04.
     runs-on: ubuntu-20.04
     strategy:
       fail-fast: false
       matrix:
-        split: ${{fromJSON(needs.matrix-gen.outputs.matrix)}}
+        class: ${{ fromJSON(needs.matrix-gen.outputs.class) }}
```
Comment (HyukjinKwon) [inline .github/workflows/benchmark.yml:111]: e.g.,  you can do sth like: `class: ${{ github.event.inputs.class }}`.

BTW, why do we define the matrix here? I think we can just:
1. Define one environment variable for the data location (e.g., `TPC_DS_DATA_LOC`) when `github.event.inputs.run-tcp-ds-query-benchmark` is true.

2. Enable TPC-DS benchmark here (https://github.com/apache/spark/pull/33544/files#diff-e50c749ae234bd48c56241408518145a665df366e62fc9978ad383a7b4afefddL99-L100) by appending the data location argument.

Comment (HyukjinKwon) [reply .github/workflows/benchmark.yml:111]: Let's also don't forget to update `object Benchmarks`'s docuentation. We can, for example, if you define `TPC_DS_DATA_LOC` environment variable with a proper data location, that triggers TPC-DS as well.

**Comment by MyeongKim** [comment]
Thank you so much for your comments. Let me test the code with your approach

**Comment by MyeongKim** [comment]
retriggered test cases with the combinations below

| input class string | run-tcp-ds-query-benchmark flag | GA url |
| ------------- | ------------- | ------------- |
| org.apache.spark.sql.* | true | https://github.com/MyeongKim/spark/actions/runs/1103168229 |
| org.apache.spark.sql.* | false | https://github.com/MyeongKim/spark/actions/runs/1103166487 |

File: .github/workflows/benchmark.yml (hunk 2 above)
Comment (MyeongKim) [reply .github/workflows/benchmark.yml]: yes. right I think your approach is simpler. removed the matrix dimension

File: .github/workflows/benchmark.yml (hunk 3 above)
Comment (MyeongKim) [reply .github/workflows/benchmark.yml]: thank you. removed the matrix dimension  and added a comment in Benchmarks object

**Comment by MyeongKim** [comment]
@HyukjinKwon Hello Hyukjin, Could you please advice me on how to proceed? Is there anything I should modify? Thank you in advance

**Comment by pingsutw** [comment]
@MyeongKim Any new progress on this? Mind me taking over this issue?

**Comment by MyeongKim** [comment]
> @MyeongKim Any new progress on this? Mind me taking over this issue?

Please go ahead!

**Comment by kazuyukitanimura** [comment]
@MyeongKim @pingsutw I opened #37020 based on this PR since I needed it for PRs I am preparing. I hope you do not mind.

**Pull request closed without merging**